lib/exsoda/http.ex

```diff
@@ -6,53 +6,90 @@ defmodule Exsoda.Http do
     Keyword.get(options, key, Config.get(:exsoda, key))
   end
 
-  def make_url(url) do
-    proto = Config.get(:exsoda, :protocol, "https")
-    api_root = Config.get(:exsoda, :api_root, "/api")
+  defp make_url(url, api_root, proto) do
     "#{proto}://#{url}#{api_root}"
   end
 
+  def base_url(%{opts: %{
+      host: {:system, env_var, default},
+      api_root: api_root,
+      protocol: protocol
+    }}) do
 
-  def base_url(%{opts: %{host: {:system, env_var, default}}}) do
     host_str = System.get_env(env_var) || default
-    {:ok, make_url(host_str)}
+    {:ok, make_url(host_str, api_root, protocol)}
   end
-  def base_url(%{opts: %{host: host}}) when is_function(host, 0) do
+  def base_url(%{opts: %{
+      host: host,
+      api_root: api_root,
+      protocol: protocol
+    }}) when is_function(host, 0) do
     with {:ok, host_str} <- host.() do
-      {:ok, make_url(host_str)}
+      {:ok, make_url(host_str, api_root, protocol)}
     end
   end
-  def base_url(%{opts: %{host: host}}) do
-    {:ok, make_url(host)}
+  def base_url(%{opts: %{host: host, api_root: api_root, protocol: protocol}}) do
```

Comment:
(addressed this in person, but recording it here)-- Mainly because the lack of when statement makes it shorter